hera_cal/utils.py

```diff
@@ -1052,66 +1055,121 @@ def red_average(hd, reds=None, bl_tol=1.0, wgt_by_int=False, inplace=False):
         inplace : bool
             Perform average and downselect inplace, otherwise returns a deepcopy.
             The first baseline in each reds sublist is kept.
+        flags : DataContainer
+            if data is a DataContainer, these are its flags
+        nsamples : DataContainer
+            if data is a DataContainer, these are its nsamples. Only required if wgt_by_int
 
     Returns:
-        HERAData : if inplace is False
+        if fed a DataContainer:
+            DataContainer, averaged data
+            DataContainer, averaged flags
+            DataContainer, summed nsamples
+        elif fed a HERAData or UVData:
+            HERAData or UVData object, averaged data
     """
-    from hera_cal import redcal
+    from hera_cal import redcal, datacontainer
+    fed_container = isinstance(data, datacontainer.DataContainer)
 
```

Comment:
We should probably raise an error if it's not a DataContainer (or a subclass of one, if we ever need one) or a UVData object (or a subclass, e.g. HERAData).

Reply:
I'd also do the checking at the top that if it's a DataContainer, `flags` and (if necessary) `nsamples` has been provided. That way, the error message can be very clear to the user.

Reply:
1. Sure thing
2. flags and nsamples are optional (even if data is a DataContainer), so there isn't such a check

Reply:
on 2, I guess that's fine. Maybe it's overkill to have a check just in the case where wgt_by_int is True.